pkg/rancher-desktop/backend/containerClient/nerdctlClient.ts

```diff
@@ -431,6 +431,45 @@ export class NerdctlClient implements ContainerEngineClient {
     }
   }
 
+  async composePort(composeDir: string, options: ContainerComposePortOptions): Promise<string> {
+    const cleanups: (() => Promise<void>)[] = [];
+
+    try {
+      const workComposeDir = await this.copyDirectoryIn(composeDir);
+
+      cleanups.push(() => this.vm.execCommand('/bin/rm', '-rf', workComposeDir));
+      const args = [
+        options.namespace ? ['--namespace', options.namespace] : [],
+        ['compose'],
+        options.name ? ['--project-name', options.name] : [],
+        ['--project-directory', workComposeDir],
+      ].flat();
+
+      if (options.env) {
+        const envFile = (await this.vm.execCommand({ capture: true },
+          '/bin/mktemp', '--tmpdir', 'rd-nerdctl-compose-exec-XXXXXX')).trim();
```

Comment:
I guess there's no need to clean this up on the VM

Reply:
The next line does the cleanup (pushing the `rm` into `cleanups`, which get run at the end of the function in a `finally`).

I'd have preferred `defer`, but that's a golang thing and not a JavaScript thing.